### ISSUE TITLE
Correct the BoundingBox.intersect(other: BoundingBox): Vector comments positions

### ISSUE THREAD

**Issue by ikudrickiy** [excaliburjs/Excalibur#3010]
### Context

All 4 multiline comments are placed incorrectly
<details>
  <summary>Screenshot</summary>
  
![image](https://github.com/excaliburjs/Excalibur/assets/139219793/bc4626d5-2302-4ad6-bd06-a6aeb317830d)
</details>

### Proposal

Correct the comments positions

**Comment by ikudrickiy** [comment]
Ah, seems like it is just another comment positioning principle. Do we have a rule to follow in this question? Should we? Shouldn't we?

**Comment by ikudrickiy** [comment]
Pay attention to this
#3011
first

**Comment by eonarheim** [comment]
@ikudrickiy Fair point! Yah these comments are confusingly placed... this could be improved. 

We'll chat about #3011 first as you say!
